src/witness/circom.rs

```diff
@@ -15,14 +15,34 @@ impl Wasm {
         Ok(())
     }
 
+
+    // Circom 2.0
+    pub fn get_version(&self) -> Result<i32> {
+        self.get_i32("getVersion")
+    }
+
+    // Circom 1
     pub fn get_fr_len(&self) -> Result<i32> {
         self.get_i32("getFrLen")
     }
 
+    // Circom 2.0
+    pub fn get_field_num_len32(&self) -> Result<i32> {
+        self.get_i32("getFieldNumLen32")
+    }
```

Comment:
You could perhaps create a `struct CircomBase`, `struct Circom` and `struct Circom2` to better separate 1.0 vs 2.0 functions, and instantiate the Wasm with one or the other based on the feature?

Reply:
I tried a few different things here, with enums/generics etc. The current trait version is the one that that seemed somewhat sensible and I managed to get to compile :sweat_smile: Let me know what you think.